src/services/tracing.ts

```diff
@@ -238,13 +238,28 @@ export const recordExceptionInCurrentSpan = ({
   recordException(span, error, level)
 }
 
+const updateErrorForSpan = ({
+  span,
+  errorLevel,
+}: {
+  span: Span
+  errorLevel: ErrorLevel
+}): boolean => {
+  const spanErrorRank = RankedErrorLevel.indexOf(span.attributes["error.level"])
+  const errorRank = RankedErrorLevel.indexOf(errorLevel)
+
+  return errorRank >= spanErrorRank
+}
+
 const recordException = (span: Span, exception: Exception, level?: ErrorLevel) => {
   // @ts-ignore-next-line no-implicit-any error
   const errorLevel = level || exception["level"] || ErrorLevel.Warn
-  span.setAttribute("error.level", errorLevel)
-  // @ts-ignore-next-line no-implicit-any error
-  span.setAttribute("error.name", exception["name"])
-  span.setAttribute("error.message", exception["message"])
+  if (updateErrorForSpan({ span, errorLevel })) {
+    span.setAttribute("error.level", errorLevel)
+    // @ts-ignore-next-line no-implicit-any error
+    span.setAttribute("error.name", exception["name"])
```

Comment:
assuming there are multiple errors, maybe we should add them here?

ie: 
```
span.setAttribute("error.name.1", exception["name"])`
span.setAttribute("error.message.1", exception["message"])
```

then:
```
span.setAttribute("error.name.2", exception["name"])`
span.setAttribute("error.message.2", exception["message"])
```


for the new error (assuming we know how many errors we are adding)

Reply:
I'm not sure if this would be useful since we also already have all the error details for multiple errors in the `Events` sections (e.g. [trace](https://ui.honeycomb.io/galoy/datasets/arvin-dev/result/tsieg7hWCGz/trace/3D9Ja83hgKA?fields%5B%5D=c_name&fields%5B%5D=c_service.name&span=954e8400b489147b))

Reply:
Done: https://github.com/GaloyMoney/galoy/pull/1574/commits/7df2168e978e9300b71a4c0d7d20e79da6d2f76e